src/index.js

```diff
@@ -0,0 +1,105 @@
+
+var {Matrix} = require('ml-matrix');
+var normalRandGenerator = require('distributions-normal-random');
+
+/**
+ * Create a dataset by the linear combination of pure elements with a specific gaussian distribution,
+ * @param {Array<Array<number>>} pureElements - matrix where each row is a pure element
+ * @param {object} options
+ * @param {boolean} options.keepCompositionMatrix - if it's true, the result object contain the composition matrix -compositionMatrix-
+ * @param {boolean} options.keepDataClass - if it's true, the result object contain the dataClass matrix.
+ * @param {boolean} options.dummyMatrix - if it's true, the dataClass matrix will be a dummyMatrix other wize it's a vector matrix.
```

Comment:
what is the purpose of this option?

Reply:
for example, you have an dataset of 5 elements and two classes, the dataClass matrix could be:
[
 [0],
 [0],
 [0],
 [1],
 [1],
]
or to have the same information as a dummy Matrix:
[
 [0, 1],
 [0, 1],
 [0, 1],
 [1, 0],
 [1, 0],
]
width dummyMatrix option you can select between those output

Reply:
Is the dummyMatrix really required by the end user?

If the answer is yes, it seems a bit odd that based on an option a property of the result has something with a different format. We should probably put it in 2 different properties, and from my understanding, the first one could be a vector instead of a matrix (it will always have only one column).

For example:
`result.classMatrix = [[0,1],[0, 1],[0, 1],[1, 0],[1, 0]];`
`result.classVector = [0, 0, 0, 1, 1];`

It would be more self-documented that way. (the end user cannot guess what `dummyMatrix` is)

Reply:
Can we remove the `keepCompositionMatrix` and keepDataClass` options? Are you afraid of cpu and memory costs?